File: 1-TwoSum/1-TwoSum.py
# Last updated: 7/28/2025, 6:15:37 PM
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        n = len(nums)
        for i in range (n-1):
            for j in range (i+1, n):
                if nums[i] + nums[j] == target :
                    return i,j
                    
        

File: 1-TwoSum/1-TwoSum.java
// Last updated: 7/28/2025, 6:15:35 PM
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> map = new HashMap<>();
        for (int i = 0; i < nums.length; i++) {
            int cur = nums[i];
            int x = target - cur;
            if (map.containsKey(x)) {
                return new int[] {map.get(x), i};
            }
            map.put(cur, i);
        }
        return null;
    }
}